File: pkg/ios/uninstaller.go
package ios

import (
	"fmt"
	"os"
	"path/filepath"
	"strings"
)

func UninstallClixIOS() error {
	appPath, err := FindAppPath()
	if err != nil {
		return err
	}
	appPath = filepath.Join(appPath, "AppDelegate.swift")
	if _, err := os.Stat(appPath); err != nil {
		return fmt.Errorf("failed to find AppDelegate.swift: %w", err)
	}

	content, err := os.ReadFile(appPath)
	if err != nil {
		return fmt.Errorf("failed to read AppDelegate.swift: %w", err)
	}

	lines := strings.Split(string(content), "\n")
	var cleaned []string

	for _, line := range lines {
		trimmed := strings.TrimSpace(line)
		if strings.Contains(trimmed, "import Clix") ||
			strings.Contains(trimmed, "import Firebase") ||
			strings.Contains(trimmed, "FirebaseApp.configure()") ||
			strings.Contains(trimmed, "Clix.initialize") ||
			strings.Contains(trimmed, "UNUserNotificationCenter.current().delegate = self") ||
			strings.HasPrefix(trimmed, "Clix.") {
			continue
		}
		cleaned = append(cleaned, line)
	}

	err = os.WriteFile(appPath, []byte(strings.Join(cleaned, "\n")), 0644)
	if err != nil {
		return fmt.Errorf("failed to write AppDelegate.swift: %w", err)
	}

	fmt.Println("✅ All the Clix SDK code has been removed from the project.")
	return nil
}

File: src/packages/ios/uninstaller.ts
import { promises as fs } from 'fs';
import path from 'path';
import { findAppPath } from './locator.js';

/**
 * Uninstalls Clix SDK from the iOS project by removing related code from AppDelegate.swift
 */
export async function uninstallClixIOS(): Promise<void> {
  const appPath = await findAppPath();
  const appDelegatePath = path.join(appPath, 'AppDelegate.swift');

  try {
    await fs.access(appDelegatePath);
  } catch (error) {
    throw new Error(`failed to find AppDelegate.swift: ${error instanceof Error ? error.message : String(error)}`);
  }

  const content = await fs.readFile(appDelegatePath, 'utf-8');
  const lines = content.split('\n');
  const cleaned: string[] = [];

  for (const line of lines) {
    const trimmed = line.trim();

    // Skip lines that contain Clix or Firebase related code
    if (
      trimmed.includes('import Clix') ||
      trimmed.includes('import Firebase') ||
      trimmed.includes('FirebaseApp.configure()') ||
      trimmed.includes('Clix.initialize') ||
      trimmed.includes('UNUserNotificationCenter.current().delegate = self') ||
      trimmed.startsWith('Clix.')
    ) {
      continue;
    }

    cleaned.push(line);
  }

  await fs.writeFile(appDelegatePath, cleaned.join('\n'), 'utf-8');
  console.log('✅ All the Clix SDK code has been removed from the project.');
}
